platforms/software/testing-base-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/TestResultProcessor.java

```diff
@@ -48,4 +49,10 @@ public interface TestResultProcessor {
      */
     @UsedByScanPlugin("test-distribution")
     void failure(Object testId, TestFailure result);
+
+    /**
+     * Notifies this processor that metadata has been emitted for the given test.
+     */
+    default void report(Object testId, TestMetadataEvent keyValueEvent) {}
```

Comment:
I think this name and comment is misleading.  The first 2 things "report" suggests to me are:

- A verb, as in "report on this test result"
- A noun, as in "the test report produced for this test"
- The other methods here are things that happened.  The test failed/completed/etc.  This would mean "The test reported", which again makes me think of the testing report produced at the end.

Also, I'm not sure it's fair to say this is only for metadata.  It could be used to transmit test params/results/etc.

How about calling this recorded? published? emitted? data?  Or even just entry?

Reply:
+1 published, that's what JUnit calls it (reportEntryPublished)